src/layout/components/NavHeader/Language.vue

```diff
@@ -41,45 +40,25 @@ export default {
       }, {})
     },
     currentLang() {
-      const lang = this.supportedLangMapper[this.currentLangCode] || this.defaultLang
-      return lang
+      const lang = getLangCode(true)
+      return this.supportedLangMapper[lang] || this.defaultLang
     }
   },
   mounted() {
-    this.currentLangCode = getLangCode()
     this.supportLanguages = store.getters.publicSettings['LANGUAGES'].map(item => {
-      let code = item.code.replace('-', '_')
-      if (code !== 'zh_hant') {
-        code = code.slice(0, 2)
-      }
       return {
         title: item.name,
-        code: code,
+        code: item.code,
         cookieCode: item.code
       }
     })
     this.changeMomentLang()
   },
   methods: {
-    changeLang() {
-      if (this.currentLang.code !== this.$i18n.locale) {
-        this.changeLangTo(this.currentLang)
-      }
-    },
     changeMomentLang() {
-      if (this.currentLang.code.indexOf('en') > -1) {
-        this.$moment.locale('en')
-        document.documentElement.lang = 'en'
-      } else if (this.currentLang.code.indexOf('ja') > -1) {
-        this.$moment.locale('ja')
-        document.documentElement.lang = 'ja'
-      } else if (this.currentLang.code.indexOf('zh_hant') > -1) {
-        this.$moment.locale('zh-tw')
-        document.documentElement.lang = 'zh-tw'
-      } else {
-        this.$moment.locale('zh-cn')
-        document.documentElement.lang = 'zh-cn'
-      }
+      const lang = getLangCode()
+      this.$moment.locale(lang)
+      document.documentElement.lang = lang
     },
     changeLangTo(item) {
       this.$axios.get(`/core/i18n/${item.cookieCode}/`).then(() => {
```

Comment:
There doesn't seem to be any significant discrepancies with the provided code snippet. There are no known issues, inconsistencies, or optimizations suggested. The code appears to be clear and functional.

The method `changeLang` is designed to adjust the current selected locale when `currentLang.code` changes. However, there does not appear to be an explicit need for it because the `$moment.locale()` function seems to work correctly on its own without being called within this same function's scope due to global namespace access. If you need help adjusting how `$moment.locale()` calls into this method behave, please specify what specific issue or behavior you're looking to address.

In summary, no major modifications were made that would indicate irregularities or inefficiencies compared to existing best practices in JavaScript coding standards at present time of writing.

If needed, I suggest checking out popular modern JavaScript libraries like Babel or TypeScript to improve readability and compatibility across different frameworks and environments.

For other concerns relevant from the context provided, consider providing more details about them so they can be appropriately addressed.